web/src/stream_create.ts

```diff
@@ -424,20 +424,29 @@ function create_stream(): void {
         },
         error(xhr): void {
             const error_message = z.object({msg: z.string().optional()}).parse(xhr.responseJSON);
-            if (error_message?.msg?.includes("access")) {
-                // If we can't access the stream, we can safely
-                // assume it's a duplicate stream that we are not invited to.
-                //
-                // BUG: This check should be using error codes, not
-                // parsing the error string, so it works correctly
-                // with i18n.  And likely we should be reporting the
-                // error text directly rather than turning it into
-                // "Error creating channel"?
+            // If we can't access the stream, we can safely
+            // assume it's a duplicate stream that we are not invited to.
```

Comment:
This is not accurate since the stream can be archived as well. Also, probably the positioning of this part is not correct.

Having `BUG` part here seems correct, but I think you can move this part below in the `else-if` block keeping the comment same for now.

Reply:
I removed complete comment since I have introduced new error_code for both the cases.